### PR TITLE
Limit some properties to use at most 63 chars

### LINKED ISSUE #487: Limit label values to 63 chars
**Description**

When an operator is defined, the length of the jaeger resource name isn't checked:

In the code, there is no check on the 63 char limit for [labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/). (Is this correct?)

> Valid label values must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.

**Question**

When deploying an operator that exceeds this limit, is there a trace of the failure?
Is there a way to fail the operator installation?
Maybe fix this problem in these ways:

* the length of the resource name is what the official specs are saying, but the operator code trims referencing labels to the label spec length?
* limit the length of the operator resource name?

### PR DESCRIPTION
**Pull request by jpkrohling**
Resolves #487 by limiting label values to 63 chars. Also included in this PR are similar manipulation for other objects, such as routes and batch jobs.

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,8 +15,8 @@ JAEGER_VERSION ?= "$(shell grep jaeger= versions.txt | awk -F= '{print $$2}')"
 OPERATOR_VERSION ?= "$(shell git describe --tags)"
 STORAGE_NAMESPACE ?= "${shell kubectl get sa default -o jsonpath='{.metadata.namespace}' || oc project -q}"
 KAFKA_NAMESPACE ?= "kafka"
-KAFKA_EXAMPLE ?= "https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/0.14.0/examples/kafka/kafka-persistent-single.yaml"
```
Comment (jpkrohling) [inline Makefile]: This version fixes a bug that caused us to use a special "managed-by" label.

File: Makefile [hunk 2]
```diff
@@ -147,7 +147,7 @@ run: crd
 
 .PHONY: run-debug
 run-debug: run
-run-debug: CLI_FLAGS = "--log-level=debug"
+run-debug: CLI_FLAGS = --log-level=debug --tracing-enabled=true
```
Comment (jpkrohling) [inline Makefile]: When running `make run-debug`, the Operator will create traces for each reconciliation loop. Just need to have a Jaeger instance at localhost.

**Comment by jpkrohling** [comment]
@kevinearls would you be able to run the tests for OpenShift ?

**Comment by kevinearls** [comment]
@jpkrohling Sure, I'll do it this afternoon

**Comment by kevinearls** [comment]
@jpkrohling I haven't run all tests so far but there are still a couple of failures just in the smoke test.  This is because your changes have broken the matching that the tests use to find routes https://github.com/jaegertracing/jaeger-operator/blob/bce10b1fe5f5cbca176f0d7f49bd287c45d3db60/test/e2e/utils.go#L363-L367

This causes the all_in_one_test/TestAllInOneWithIngress to fail as the route it creates looks like this:  

NAME                                         HOST/PORT                                                                                                   PATH   SERVICES                       PORT    TERMINATION   WILDCARD
route.route.openshift.io/my-jaeger-with-in   my-jaeger-with-in-osdk-e2e-7ecebb2f-8306-4f47-a0bc-49fb36c771e0.apps.ocp4-jqe4.maistra.upshift.redhat.com          my-jaeger-with-ingress-query   <all>   edge          None

So we need a better way to find routes for tests that use ingress.  If you have any good suggestions I can try them.

**Comment by jpkrohling** [comment]
You can use a similar code to what I'm using to define the route's name:

https://github.com/jaegertracing/jaeger-operator/blob/b445b162ee150c34a4dab1fd1b98684444eb6cf2/pkg/route/query.go#L43

**Comment by kevinearls** [comment]
Hi @jpkrohling  I did another test run this morning, and some tests are fixed, but anything using a self provisioning es instance still fails.  I created a project named and deployed the deploy/examples/openshift/simple-prod-deploy-es.yaml.  It never deploys, and I get errors including the following message from the ES operator.

Invalid value: \"elasticsearch-cdm-osdke2ee7864afba6854e498f316bd37347f666simpleprod-1\": must be no more than 63 characters

**Comment by jpkrohling** [comment]
I'll try it out. Thanks for testing!

**Comment by kevinearls** [comment]
@jpkrohling I have rerun the tests (with my changes) and everything passed except the new sidecar_namespace_test.   I think that is a separate issue, so I'd recommend you go ahead and merge this.  

Please ping me when you do, as then I will need to submit a PR with a couple of test changes.

**Comment by rubenvp8510** [comment]
@jpkrohling  @kevinearls 

While I was working on another PR I ran the tests locally and the sidecar_namespace_test fails for me too, I took a look and the reason is because the route status is rejected:

```
NAME                         HOST/PORT     PATH      SERVICES                           PORT      TERMINATION   WILDCARD
agent-as-sidecar-namespace   InvalidHost             agent-as-sidecar-namespace-query   <all>     edge          None
```

The error is this:

```
        message: 'host name validation errors: spec.host: Invalid value: "agent-as-sidecar-namespace-osdk-e2e-d60d5f9e-5713-406e-81c6-261d149c26a6.apps-crc.testing":
          must be no more than 63 characters'
        reason: InvalidHost
```

It might be related to the error you are fixing in this PR.

**Comment by jpkrohling** [comment]
@rubenvp8510 said:
> It might be related to the error you are fixing in this PR.

Yes, that's what we are fixing here.

Thanks @kevinearls, I'm merging this.

**Pull request merged**